openhcl/underhill_attestation/src/igvm_attest/key_release.rs

```diff
@@ -190,14 +196,20 @@ impl AkvKeyReleaseJwtHelper {
         let header = base64::engine::general_purpose::URL_SAFE_NO_PAD
             .decode(header)
             .map_err(AkvKeyReleaseJwtError::DecodeBase64UrlJwtHeader)?;
-        let header = String::from_utf8_lossy(&header);
+        let header = std::str::from_utf8(&header).map_err(|_| {
```

Comment:
Also preserve the error of `from_utf8`

Reply:
Let me know if the changes in https://github.com/microsoft/openvmm/pull/583/commits/0769ddcf6aedc5b1c9a92ae3fb01133e6273f1c7 don't satisfy the purpose of capturing `Utf8Error` here. My interpretation is that `Utf8Error` would allow us to inspect the invalid UTF-8 bytes encountered, but I'm including the raw bytes directly in the error message so I don't know what else `Utf8Error` would give us (but please chime in if there's value I'm missing).

Reply:
We generally prefer the original error especially from 3rd party crate over customized ones for better maintainability. I haven't look into the Utf8Error, but please document the reason if you think the customized one is better.